source/base/parameter_handler.cc

```diff
@@ -1086,6 +1102,13 @@ ParameterHandler::set(const std::string &entry_string,
 
       // finally write the new value into the database
       entries->put(path + path_separator + "value", new_value);
+      entries->put(path + path_separator + "has_been_set", true);
+
+      entries_set_status.find(path)->second = std::pair<bool, bool>(
+        entries->get<std::string>(path + path_separator + "has_to_be_set") ==
+          "true",
+        entries->get<std::string>(path + path_separator + "has_been_set") ==
+          "true");
```

Comment:
Then the following code could be simplified to:
```suggestion
      entries->put(path + path_separator + "has_been_set", true);
```